tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp

```diff
@@ -142,143 +206,185 @@ std::shared_ptr<trajopt::ProblemConstructionInfo> TrajOptPlannerDefaultConfig::g
 
     pci.cnt_infos.insert(pci.cnt_infos.end(), term_info.cnt.begin(), term_info.cnt.end());
     pci.cost_infos.insert(pci.cost_infos.end(), term_info.cost.begin(), term_info.cost.end());
-  }
 
-  /* Update the first and last step for the costs
-   * Certain costs (collision checking and configuration) should not be applied to start and end states
-   * that are incapable of changing (i.e. joint positions). Therefore, the first and last indices of these
-   * costs (which equal 0 and num_steps-1 by default) should be changed to exclude those states
-   */
-  int cost_first_step = 0;
-  int cost_last_step = pci.basic_info.n_steps - 1;
-  if (target_waypoints.front()->getType() == WaypointType::JOINT_WAYPOINT ||
-      target_waypoints.front()->getType() == WaypointType::JOINT_TOLERANCED_WAYPOINT)
-  {
-    ++cost_first_step;
+    /* Update the first and last step for the costs
+     * Certain costs (collision checking and configuration) should not be applied to start and end states
+     * that are incapable of changing (i.e. joint positions). Therefore, the first and last indices of these
+     * costs (which equal 0 and num_steps-1 by default) should be changed to exclude those states
+     */
+    if (target_waypoints[ind]->getType() == WaypointType::JOINT_WAYPOINT)
+    {
+      fixed_steps.push_back(static_cast<int>(ind));
+    }
   }
-  if (target_waypoints.back()->getType() == WaypointType::JOINT_WAYPOINT ||
-      target_waypoints.back()->getType() == WaypointType::JOINT_TOLERANCED_WAYPOINT)
+}
+
+void TrajOptPlannerDefaultConfig::addConfiguration(trajopt::ProblemConstructionInfo& pci,
```

Comment:
Perhaps we should make this addConfigurationCost or addKinematicConfiguration. addConfiguration sounds like you are adding a "config".

Reply:
Sure.

Reply:
done